src/Components/Table/LogLinePill.tsx

```diff
@@ -0,0 +1,163 @@
+import React, { useMemo } from 'react';
+import { Row } from 'react-table';
+import { css, cx } from '@emotion/css';
+
+import { DataFrame, Field, FieldType, getLinksSupplier, GrafanaTheme2, LinkModel } from '@grafana/data';
+import { getCellLinks, useTheme2 } from '@grafana/ui';
+
+import { useTableCellContext } from 'Components/Table/Context/TableCellContext';
+import { CellContextMenu } from 'Components/Table/CellContextMenu';
+import { getFieldMappings } from 'Components/Table/Table';
+import { FieldNameMetaStore } from 'Components/Table/TableTypes';
+import { useTableColumnContext } from 'Components/Table/Context/TableColumnsContext';
+import { getTemplateSrv } from '@grafana/runtime';
+
+interface LogLinePillProps {
+  originalField?: Field;
+  field?: Field;
+  columns: FieldNameMetaStore;
+  label: string;
+  showColumns: () => void;
+  rowIndex: number;
+  frame: DataFrame;
+  originalFrame: DataFrame | undefined;
+  isDerivedField: boolean;
+  value: string;
+}
+
+const getStyles = (theme: GrafanaTheme2, bgColor?: string) => ({
+  pill: css({
+    flex: '0 1 auto',
+    marginLeft: '5px',
```

Comment:
Nit, but many of these could be `spacing()`.

Reply:
good catch, cleaned up in latest push